test/boost/sstable_test.cc

```diff
@@ -166,17 +166,12 @@ SEASTAR_TEST_CASE(missing_summary_first_last_sane) {
 }
 
 static future<sstable_ptr> do_write_sst(test_env& env, schema_ptr schema, sstring load_dir, sstring write_dir, sstables::generation_type generation) {
-    return env.reusable_sst(std::move(schema), load_dir, generation).then([write_dir, generation] (sstable_ptr sst) mutable {
-        sstable_generation_generator gen{generation.as_int()};
-        sstables::test(sst).change_generation_number(gen());
-        auto fut = sstables::test(sst).change_dir(write_dir);
-        return std::move(fut).then([sst = std::move(sst)] {
-            auto fut = sstables::test(sst).store();
-            return std::move(fut).then([sst = std::move(sst)] {
-                return make_ready_future<sstable_ptr>(std::move(sst));
-            });
-        });
-    });
+    auto sst = co_await env.reusable_sst(std::move(schema), load_dir, generation);
+    sstable_generation_generator gen(generation.as_int());
+    sstables::test(sst).change_generation_number(gen());
+    co_await sstables::test(sst).change_dir(write_dir);
+    co_await sstables::test(sst).store();
+    co_return std::move(sst);
```

Comment:
nit: this std::move is probably unneeded, but harmless either way.

Reply:
Actually, we prefer copy-elision to take place in `co_return sst`.
The compiler doesn't help us in `return make_ready_future<shared_sstable>(sst);` because we're making an rvalue, but in this case we're not, so it's better to leave it for the compiler to elide.

Reply:
@bhalevy 

> copy-elision to take place in `co_return sst`

NRVO could apply -- dependent on the compiler's decision -- if this were `return sst`. But it's not; it's `co_return sst`. `co_return` is [not subject to NRVO](https://stackoverflow.com/questions/61870280/does-nrvo-also-apply-to-coroutines/72513585#72513585), as far as I understand. `co_return` amounts to a function call to `coroutine_traits_base::return_value()`, so NRVO is off the table; the best we can do is move, with `co_return`, AIUI.

Reply:
IOW, I agree that `co_return sst` is good -- but not because it can elide copy or move. Instead (IMO), `co_return sst` is good because it's simpler to read than `co_return std::move(sst)`, and *still* makes `sst` move-eligible.